src/scale.jl

```diff
@@ -251,45 +253,52 @@ function apply_scale_typed!(ds, field, scale::ContinuousScale)
 end
 
 # Reorder the levels of a pooled data array
-function reorder_levels(da::PooledDataArray, order::AbstractVector)
-    level_values = levels(da)
+function reorder_levels(da::IndirectArray, order::AbstractVector)
+    level_values = da.values
     length(order) != length(level_values) &&
             error("Discrete scale order is not of the same length as the data's levels.")
     permute!(level_values, order)
-    return PooledDataArray(da, level_values)
+    return IndirectArray(da, level_values)
 end
 
-function discretize_make_pda(values::Vector, levels=nothing)
-    if levels == nothing
-        return PooledDataArray(values)
-    else
-        return PooledDataArray(convert(Vector{eltype(levels)}, values), levels)
-    end
-end
+discretize_make_ia(values::Vector)         = IndirectArray(values)
+discretize_make_ia(values::Vector, ::Void) = IndirectArray(values)
+discretize_make_ia(values::Vector, levels) =
```

Comment:
`IndirectArray`s behave differently from PDAs if some levels are omitted. Do you think this is used anywhere (i.e. to skip some levels)?
```julia
julia> PooledDataArray(["a", "b", "c"], ["a", "c"])
3-element DataArrays.PooledDataArray{String,UInt32,1}:
 "a"    
 missing
 "c"    

julia> IndirectArray(["a", "b", "c"], ["a", "c"])
ERROR: BoundsError: attempt to access 2-element Array{String,1} at index [UInt8[0x01, 0x00, 0x02]]
```

Reply:
I'm not completely sure but it looks like Gadfly typically constructs the union of available levels. If Gadfly made use of this behavior, I'd also expect that at least one of the tests would have caught it.

Reply:
Yeah, I guess we'll soon find out if that's not the case anyway. :-)